src/iface/fragmentation.rs

```diff
@@ -0,0 +1,555 @@
+#![allow(unused)]
+
+use managed::{ManagedMap, ManagedSlice};
+
+use crate::storage::Assembler;
+use crate::time::Instant;
+use crate::Error;
+use crate::Result;
+
+pub trait PacketAssemblerInfo: PartialEq {
+    /// Calculate a new offset based on some other information.
+    fn calc_offset(&self, offset: usize) -> usize;
```

Comment:
Why is this needed? from the 6lowpan code it seems it adds a fixed offset, could the calling code do that on its own instead?